### DISCUSSION TITLE
Suggestion: Three.js example on on homepage: https://elalish.github.io/manifold/

### DISCUSSION THREAD

**Discussion by bhouston** [elalish/manifold#117] [Ideas]
Once you have a Three.js example, it would be nice to feature it on the homepage of manifold with some type of impressive animation.  Two objects moving relatively to each other and showing the complex intersection being resolved by this library in real time.  That would immediately sell people on the value of this library.

**Comment by elalish** [comment]
Again, I think you miss the point of what this library does; it's not intended to be a real-time solution. It's intended to make very complicated designs possible without fear of crashing the CAD program or taking forever to generate.

**Comment by bhouston** [comment]
What is the performance of this library?  Is there numbers you can share?

**Comment by FishOrBear** [comment]
I think we are all evaluating whether this library can be compiled to wasm and then used on a web 3d editor.

**Comment by elalish** [comment]
That would be great. Do you have any experience with building WASM?

**Comment by fire** [comment]
It should be technically possible to compile the library to wasm and then execute from three js rendering mesh buffers.

@elalish Did you remove the CUDA dependency?

**Comment by elalish** [comment]
CUDA is not exactly a dependency; this library uses Nvidia's Thrust which can target either CUDA, OMP, or standard C++ backends. For instance, the tests in Github actions all use the C++ backend as they don't give us machines with GPUs. A CMake flag chooses the backend you wish to build. 

It may be possible to build the C++ version as WASM, though it would be single-threaded (still pretty fast though). Long term, I think the ideal thing would be to build WASM using WebGPU compute kernels. I would imagine someone is working on making C++17 parallel algorithms (which is based on and very similar to Thrust) compile to WASM with WebGPU, but it hasn't happened yet as far as I know. I would rather wait for a compiler stage like that to exist than try to jump right onto the bleeding edge myself.
